src/rust/engine/src/cffi_build.rs

```diff
@@ -18,10 +18,36 @@ native engine binary, allowing us to address it both as an importable python mod
 
 */
 
+use std::env;
+use std::fs;
+use std::io;
+use std::io::Read;
+
 fn main() {
-  gcc::Config::new()
-    // N.B. The filename of this source code - at generation time - must line up 1:1 with the
-    // python import name, as python keys the initialization function name off of the import name.
-    .file("src/cffi/native_engine.c")
-    .compile("libnative_engine_ffi.a");
+  let mut config = gcc::Config::new();
+
+  // N.B. The filename of this source code - at generation time - must line up 1:1 with the
+  // python import name, as python keys the initialization function name off of the import name.
+  config.file("src/cffi/native_engine.c");
+  for flag in make_flags().unwrap() {
+    config.flag(flag.as_str());
+  }
+
+  config.compile("libnative_engine_ffi.a");
+}
+
+fn make_flags() -> io::Result<Vec<String>> {
+  let env_file_path = {
+    let mut path = env::current_dir().unwrap(); // ${PANTS_ROOT}/src/rust/engine
+    path.push("src");
```

Comment:
The portability of using `push` with individual path components is nice, but it might be more valuable to extract the two "literal" paths used in this file into statics:
```
static CFLAGS_PATH: &'static str = "src/cffi/native_engine.cflags";
static C_PATH: &'static str = "src/cffi/native_engine.c";
```
...so it's more obvious that they're intended to correspond.